queries/usd/highlights.scm

```diff
@@ -0,0 +1,228 @@
+(None) @constant.builtin
+(asset_path) @string.special.asset
+(bool) @boolean
+(comment) @comment @spell
+(custom) @function.builtin
+(float) @float
+(integer) @number
+(namespace_identifier) @namespace
+(orderer) @function.call
+(prim_path) @string.special.prim
+(uniform) @function.builtin
+
+[
+  "class"
+  "def"
+  "over"
+] @keyword.function
+
+["(" ")" "[" "]" "{" "}"] @punctuation.bracket
+[":" ";" "."] @punctuation.delimiter
+
+[
+  "="
+] @operator
+
+;; Simple variables
+(attribute_assignment
+ (identifier) @variable
+)
+(attribute_declaration
+ (identifier) @variable
+)
+(relationship_declaration
+ (identifier) @variable
+)
+(relationship_assignment
+ (identifier) @variable
+)
+(qualified_identifier
+  (identifier) @variable)
+
+;; Attribute components
+(attribute_property) @property
+
+(attribute_type) @type
+(
+ ;; Reference: https://openusd.org/release/api/sdf_page_front.html
+ (attribute_type) @type.builtin
+ (#any-of? @type.builtin
+  ;; Scalar types
+  "asset" "asset[]"
+  "bool" "bool[]"
+  "double" "double[]"
+  "float" "float[]"
+  "half" "half[]"
+  "int" "int[]"
+  "int64" "int64[]"
+  "string" "string[]"
+  "timecode" "timecode[]"
+  "token" "token[]"
+  "uchar" "uchar[]"
+  "uint" "uint[]"
+  "uint64" "uint64[]"
+
+  ;; Dimensioned Types
+  "double2" "double2[]"
+  "double3" "double3[]"
+  "double4" "double4[]"
+  "float2" "float2[]"
+  "float3" "float3[]"
+  "float4" "float4[]"
+  "half2" "half2[]"
+  "half3" "half3[]"
+  "half4" "half4[]"
+  "int2" "int2[]"
+  "int3" "int3[]"
+  "int4" "int4[]"
+  "matrix2d" "matrix2d[]"
+  "matrix3d" "matrix3d[]"
+  "matrix4d" "matrix4d[]"
+  "quatd" "quatd[]"
+  "quatf" "quatf[]"
+  "quath" "quath[]"
+
+  ;; Extra Types
+  "color3f" "color3f[]"
+  "normal3f" "normal3f[]"
+  "point3f" "point3f[]"
+  "texCoord2f" "texCoord2f[]"
+  "vector3d" "vector3d[]"
+  "vector3f" "vector3f[]"
+  "vector3h" "vector3h[]"
+
+  "dictionary"
+
+  ;; Deprecated Types
+  "EdgeIndex" "EdgeIndex[]"
+  "FaceIndex" "FaceIndex[]"
+  "Matrix4d" "Matrix4d[]"
+  "PointIndex" "PointIndex[]"
+  "PointFloat" "PointFloat[]"
+  "Transform" "Transform[]"
+  "Vec3f" "Vec3f[]"
+ )
+)
+
+;; Relationships
+(relationship_type) @type
+(relationship_assignment
+  (identifier) @variable
+)
+(relationship_declaration
+  (identifier) @variable
+)
+
+(variant_set_definition) @keyword
+
+;; String rules
+(attribute_assignment
+ (string) @string)
+(dictionary
+ (string) @string)
+(list
+ (string) @string)
+; In USD def "foo" ("This is a docstring") {} < the ""s within the ()s is not
+; a string but a docstring
+;
+(metadata
+ (comment)*
+ .
+ (string) @comment.documentation)
+(metadata_assignment
+ (string) @string)
+(prim_definition
+ (string) @string)
+(variant_set_definition
+ (string) @string)
+(variant
+ (string) @string)
```

Comment:
```(string) @string```, no need for all this

Reply:
for the section 
```scheme
(metadata
 (comment)*
 .
 (string) @comment.documentation)
```

The later it appears in the file, the higher priority it is

Reply:
Just copying this comment here, because it's related

https://github.com/nvim-treesitter/nvim-treesitter/pull/4774#discussion_r1188790357

"foo" - A regular string. Depending on where in the file this is defined, this is actually a raw string, a docstring, or a comment

In USD, what we commonly think of as a string might be a string defining a docstring, a raw string, or a comment depending on where it's located in the file. My intent here was to be explicit about each case so that there's less chance of conflicting highlight groups matching. If I can express that intent more concisely please let me know

Reply:
> The later it appears in the file, the higher priority it is

Oh okay, I didn't know that. I'll reorder the highlights to reflect that. Currently in a couple places it's the opposite order than it should be

Reply:
Confirmed these edits + moving the (metadata ... (string) ...) to the end of the file passes unittests. Done!